SVG rendering ignores xml:space="preserve" attribute for text

Ahmad-S792 opened the pull request:
#### e4359d3e33a4927ce8c7164b4d277693ddfa1806
<pre>
SVG rendering ignores xml:space=&quot;preserve&quot; attribute for text

<a href="https://bugs.webkit.org/show_bug.cgi?id=112032">https://bugs.webkit.org/show_bug.cgi?id=112032</a>
<a href="https://rdar.apple.com/problem/19164263">rdar://problem/19164263</a>

Reviewed by Nikolas Zimmermann.

This patch aligns WebKit with Gecko / Firefox and Blink / Chromium.

Merge: <a href="https://source.chromium.org/chromium/chromium/src/+/66a66b7e89c9cb4d2f0579354f8f84a202dce332">https://source.chromium.org/chromium/chromium/src/+/66a66b7e89c9cb4d2f0579354f8f84a202dce332</a>

Web Specification: <a href="https://svgwg.org/svg2-draft/text.html#LegacyXMLSpace">https://svgwg.org/svg2-draft/text.html#LegacyXMLSpace</a>

It landed in past 255917@main but led to regression and was reverted in 258905@main.
Prior to relanding, I tested the regression `chart testcase` attached on
bug 250603 and it was working as intended and didn&apos;t had &apos;garbled&apos; text
anymore.

&quot;It will convert all newline and tab characters into space characters. Then,
it will draw all space characters, including leading, trailing and
multiple contiguous space characters...&quot;

* Source/WebCore/css/svg.css:
(text):
(tspan, textPath):
(text, tspan, textPath): Deleted.
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/svg/import/text-tspan-02-b-manual-expected.txt:
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/svg/import/text-tspan-02-b-manual-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/svg/import/text-tspan-02-b-manual-expected.txt:

Canonical link: <a href="https://commits.webkit.org/281364@main">https://commits.webkit.org/281364@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/6a01dcc2a3e136b2acf2af78ae46bf8d80045f5e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/59155 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/38498 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/11668 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/63117 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/9598 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/61284 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/46152 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/9827 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/48104 "") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/6859 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/61185 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/35993 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/51201 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/28929 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/32715 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/8462 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/8602 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/54671 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/8743 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/64822 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/3083 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/8693 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/55432 "") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/3094 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/51199 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/55530 "Passed tests") | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/19/builds/13234 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/2574 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/34314 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/35397 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/36483 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/35142 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on Source/WebCore/css/svg.css:
```diff
@@ -61,8 +61,12 @@ text, foreignObject {
+
+tspan, textPath {
+    white-space: inherit;
 }
```
nit: inconsistent indentation

Ahmad-S792 replied:
@nt1m - fixed in https://github.com/WebKit/WebKit/commit/6a01dcc2a3e136b2acf2af78ae46bf8d80045f5e.

nikolaszimmermann reviewed (approved):
LGTM, r=me.

Commit:
SVG rendering ignores xml:space="preserve" attribute for text

https://bugs.webkit.org/show_bug.cgi?id=112032
rdar://problem/19164263

Reviewed by Nikolas Zimmermann.

This patch aligns WebKit with Gecko / Firefox and Blink / Chromium.

Merge: https://source.chromium.org/chromium/chromium/src/+/66a66b7e89c9cb4d2f0579354f8f84a202dce332

Web Specification: https://svgwg.org/svg2-draft/text.html#LegacyXMLSpace

It landed in past 255917@main but led to regression and was reverted in 258905@main.
Prior to relanding, I tested the regression `chart testcase` attached on
bug 250603 and it was working as intended and didn't had 'garbled' text
anymore.

"It will convert all newline and tab characters into space characters. Then,
it will draw all space characters, including leading, trailing and
multiple contiguous space characters..."

* Source/WebCore/css/svg.css:
(text):
(tspan, textPath):
(text, tspan, textPath): Deleted.
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/svg/import/text-tspan-02-b-manual-expected.txt:
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/svg/import/text-tspan-02-b-manual-expected.txt:
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/svg/import/text-tspan-02-b-manual-expected.txt:

Canonical link: https://commits.webkit.org/281364@main
